app/controllers/course_roster_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CourseRosterController < ApplicationController
```

Comment:
The name should be plural: `CourseRostersController` and same with the file: `course_rosters_controller.rb`

Perhaps you ran a generator wrong?